### PR TITLE
Python Ice/async demo improvements

### PR DESCRIPTION
**Pull request by bentoi**

### CODE REVIEW

File: python/Ice/async/client.py [hunk 1]
```diff
@@ -22,62 +22,51 @@ async def main():
...
+                elif c == 'd':
+                    # create asyncio task from coroutine returned by slowSayHello
+                    asyncio.create_task(slowSayHello(hello))
+                elif c == 's':
+                    # create asyncio future from Ice future
+                    await Ice.wrap_future(hello.shutdownAsync())
+                elif c == '?':
+                    menu()
+                else:
+                    print(f"unknown command {c!r}")
```
Comment (bernardnormier) [inline python/Ice/async/client.py:53]: The 'x' handling was lost.

Then, when we press 'x', should we wait for all outstanding 'd' tasks to complete? Or cancel them? Currently it just exits immediately.

File: python/Ice/async/client.py [hunk 2]
```diff
@@ -22,62 +22,51 @@ async def main():
...
-
-
-async def getNextChar():
-    # run blocking IO with default executor
-    c = await asyncio.get_running_loop().run_in_executor(None, sys.stdin.readline)
-    return c.strip()
+        c = None
+        while c != 'x':
+            try:
+                sys.stdout.write("==> ")
```
Comment (bernardnormier) [inline python/Ice/async/client.py:35]: Should we instead
```
print("==> ", end='', flush=True)
```
?

**Review by bernardnormier — APPROVED** [review]

File: python/Ice/async/client.py (hunk 1 above)
Comment (bentoi) [reply python/Ice/async/client.py:53]: I'm not sure it makes a big difference given that we're exiting but I added back the cancellation of pending tasks. I also added code to ensure we wait for these tasks to complete and handle the CancelledError from the coroutine.

File: python/Ice/async/client.py (hunk 2 above)
Comment (bentoi) [reply python/Ice/async/client.py:35]: Either if fine with me as long as it still works with our expect demo scripts. If we decide to change this, it's best to do this for all demos separately from this PR.

File: python/Ice/async/client.py [hunk 3]
```diff
@@ -15,69 +15,71 @@ async def main():
...
-                    except RuntimeError as ex:
-                        print("task runtime error:", ex)
-
-
-async def getNextChar():
-    # run blocking IO with default executor
-    c = await asyncio.get_running_loop().run_in_executor(None, sys.stdin.readline)
-    return c.strip()
+                        await asyncio.wait(tasks)
+                    pass
```
Comment (bernardnormier) [inline python/Ice/async/client.py:59]: Do you need this pass?

Comment (bentoi) [reply python/Ice/async/client.py]: Nope, it's a left over.

**Pull request merged**